### PR TITLE
Implement the v5 settings schema

### LINKED ISSUE #3746: Implement settings v5 schema [2d]
The v5 structure layout is being discussed in #3740 and the final version should be copied over into this story, once the discussion is settled.

This story is basically just a bunch of variable renames.

Please put the move from `kubernetes.containerEngine` to `containerEngine.Name` into a separate commit, because only a subset of the `containerEngine` references are affected, so this will be easier to review.

# Acceptance Criteria

| Scenario | Expected Output | 
| :---: | :---: |
|  Fresh install Rancher Desktop | check schema of created settings.json  |

### PR DESCRIPTION
**Pull request by ericpromislow**
Fixes #3746 and #3748

### CODE REVIEW

File: e2e/rdctl.e2e.spec.ts [hunk 1]
```diff
@@ -222,13 +222,16 @@ test.describe('Command server', () => {
...
+      containerEngine: {
+        name:           { desiredEngine },
+        imageAllowList: { locked: !settings.containerEngine.imageAllowList.locked },
+      },
       kubernetes:
         {
           enabled:                    desiredEnabled,
-          containerEngine:            desiredEngine,
           version:                    desiredVersion,
           checkForExistingKimBuilder: !settings.kubernetes.checkForExistingKimBuilder, // not supported
```
Comment (jandubois) [inline e2e/rdctl.e2e.spec.ts:236]: I thought all references to `kim` are supposed to be removed by this PR?

```console
$ agi kimbuild
e2e/rdctl.e2e.spec.ts
236:          checkForExistingKimBuilder: !settings.kubernetes.checkForExistingKimBuilder, // not supported

pkg/rancher-desktop/config/settings.ts
399:    // settings.kubernetes.checkForExistingKimBuilder = true;

pkg/rancher-desktop/main/commandServer/__tests__/settingsValidator.spec.ts
67:        ['kubernetes', 'checkForExistingKimBuilder'],

scripts/check-api-schema.ts
58:  'kubernetes.checkForExistingKimBuilder',
```

Comment (ericpromislow) [reply e2e/rdctl.e2e.spec.ts]: good point...

File: pkg/rancher-desktop/config/settings.ts [hunk 2]
```diff
@@ -391,9 +394,12 @@ const updateTable: Record<number, (settings: any) => void> = {
...
   },
-  3: (settings) => {
-    // Should stay true until the kim-based buildkit artifacts are removed -- see code in lima.ts:start()
-    settings.kubernetes.checkForExistingKimBuilder = true;
+  3: (_) => {
+    // With settings v5, all traces of the kim builder are gone now.
+    // settings.kubernetes.checkForExistingKimBuilder = true;
+  },
+  4: (_) => {
+    console.log(`TODO: Implement this!`);
```
Comment (jandubois) [inline pkg/rancher-desktop/config/settings.ts:402]: Do we have a story for this?

**Review by jandubois — APPROVED** [review]
LGTM, thanks!

I have a hard time following all the different `adminAccess` variables, but i think this is fine.

File: pkg/rancher-desktop/config/settings.ts (hunk 2 above)
Comment (ericpromislow) [reply pkg/rancher-desktop/config/settings.ts:402]: #3747

**Pull request merged**